### PR TITLE
[SELC-4645] Fix: added check to insitution.onboarding in createUser and addUserProductRole

### PR DESCRIPTION
**Pull request by flaminiaScarciofolo**
#### List of Changes

- Added check to insitution.onboarding in createUser and addUserProductRole

#### Motivation and Context

When we createNewUser we need to check if product is already onboarded on the institution

#### How Has This Been Tested?

local env

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/UserConnectorImpl.java [hunk 1]
```diff
@@ -256,4 +223,29 @@ private it.pagopa.selfcare.user.generated.openapi.v1.dto.User buildUser(UserToCr
                 .institutionEmail(userDto.getEmail())
                 .build();
     }
+    private Product1 buildProduct(String productId, List<it.pagopa.selfcare.dashboard.connector.model.user.CreateUserDto.Role> roles) {
+        return Product1.builder()
+                .productRole(roles.stream().map(it.pagopa.selfcare.dashboard.connector.model.user.CreateUserDto.Role::getProductRole).toList())
```
Comment (gunzip) [inline connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/UserConnectorImpl.java:228]: what if roles is empty?

File: core/src/main/java/it/pagopa/selfcare/dashboard/core/UserV2ServiceImpl.java [hunk 2]
```diff
@@ -133,10 +134,21 @@ public String createUsers(String institutionId, String productId, UserToCreate u
...
+        verifyOnboardingStatus(institutionId, productId);
         List<CreateUserDto.Role> role = retrieveRole(productId, productRoles);
         userApiConnector.createOrUpdateUserByUserId(institutionId, productId, userId, role);
         log.trace("createOrUpdateUserByUserId end");
     }
+
+    private void verifyOnboardingStatus(String institutionId, String productId) {
+        Institution institution = msCoreConnector.getInstitution(institutionId);
+        if (institution.getOnboarding() == null || institution.getOnboarding().stream()
+                .noneMatch(onboarding -> onboarding.getProductId().equals(productId) && onboarding.getStatus().equals(RelationshipState.ACTIVE))
```
Comment (gunzip) [inline core/src/main/java/it/pagopa/selfcare/dashboard/core/UserV2ServiceImpl.java:146]: might a comment here help clarify this condition?

File: connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/UserConnectorImpl.java (hunk 1 above)
Comment (flaminiaScarciofolo) [reply connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/UserConnectorImpl.java:228]: We check if roles is empty on line 200

File: core/src/main/java/it/pagopa/selfcare/dashboard/core/UserV2ServiceImpl.java (hunk 2 above)
Comment (empassaro) [reply core/src/main/java/it/pagopa/selfcare/dashboard/core/UserV2ServiceImpl.java:151]: added

File: connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/UserConnectorImpl.java [hunk 3]
```diff
@@ -256,4 +222,34 @@ private it.pagopa.selfcare.user.generated.openapi.v1.dto.User buildUser(UserToCr
...
+
+        if(roles == null || roles.isEmpty()) {
+            throw new IllegalArgumentException("Role list cannot be empty");
+        }
+
+        AddUserRoleDto addUserRoleDto = AddUserRoleDto.builder()
+                .institutionId(institutionId)
+                .product(Product.builder()
+                        .productRole(roles.stream().map(it.pagopa.selfcare.dashboard.connector.model.user.CreateUserDto.Role::getProductRole).toList())
+                        .role(it.pagopa.selfcare.user.generated.openapi.v1.dto.PartyRole.valueOf(roles.get(0).getPartyRole().name()))
```
Comment (manuraf) [inline connector/rest/src/main/java/it/pagopa/selfcare/dashboard/connector/rest/UserConnectorImpl.java:246]: why do we get the first roles? you could add some comments about

**Review by manuraf — CHANGES_REQUESTED** [review]

**Review by manuraf — APPROVED** [review]

**Pull request merged**